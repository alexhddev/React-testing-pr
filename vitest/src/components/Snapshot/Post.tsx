

export function Post(props: {
    content: string,
    user: string
}){

    return <div data-testid="post-container">
        <h2>{props.user}:</h2>
        <p>{props.content}</p>
        <p>Footer</p>
    </div>

}