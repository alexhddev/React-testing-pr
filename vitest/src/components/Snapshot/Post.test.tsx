import { Post } from './Post'
import { render, } from "@testing-library/react"

describe('Post snapshot tests', ()=>{

    it('initial test', ()=>{
        const rendered = render(<Post
            content='Some Content'
            user='Some user'
        />)
        expect(rendered.asFragment()).toMatchSnapshot();
    })

    it('Second test', ()=>{
        const rendered = render(<Post
            content='Some Other Content'
            user='Some Other user'
        />)
        expect(rendered.asFragment()).toMatchSnapshot();
    })

})