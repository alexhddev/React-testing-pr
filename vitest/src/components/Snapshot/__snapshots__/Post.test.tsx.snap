// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Post snapshot tests > Second test 1`] = `
<DocumentFragment>
  <div
    data-testid="post-container"
  >
    <h2>
      Some Other user:
    </h2>
    <p>
      Some Other Content
    </p>
    <p>
      Footer
    </p>
  </div>
</DocumentFragment>
`;

exports[`Post snapshot tests > initial test 1`] = `
<DocumentFragment>
  <div
    data-testid="post-container"
  >
    <h2>
      Some user:
    </h2>
    <p>
      Some Content
    </p>
    <p>
      Footer
    </p>
  </div>
</DocumentFragment>
`;
